playground/PostgresEndToEnd/PostgresEndToEnd.ApiService/NpgsqlDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,121 @@
+using System.Text;
+using System.Text.Json;
+using Azure.Core;
+using Azure.Identity;
+using Npgsql;
+
+/// <summary>
+/// Extension methods for NpgsqlDataSourceBuilder to enable Entra authentication with Azure DB for PostgreSQL.
+/// This class provides methods to configure NpgsqlDataSourceBuilder to use Entra authentication, handling token
+/// acquisition and connection setup. It is not specific to this repository and can be used in any project that
+/// requires Entra authentication with Azure DB for PostgreSQL.
+/// </summary>
+/// <example>
+/// Example usage:
+/// <code>
+/// var dataSourceBuilder = new NpgsqlDataSourceBuilder("{connection string}");
+/// dataSourceBuilder.UseEntraAuthentication();
+/// var dataSource = dataSourceBuilder.Build();
+/// </code>
+/// </example>
+public static class NpgsqlDataSourceBuilderExtensions
```

Comment:
This file is only temporary until [Consider building an Azure Postgres client integration (dotnet/aspire#6669)](https://github.com/dotnet/aspire/issues/6669) is completed.